pkg/cloudprovider/utils.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+const (
+	defaultLBSourceRange = "0.0.0.0/0"
+)
+
+// TODO: Should I move this into utils/sets ?  Even though it isn't generated?
+type IPNetSet map[string]*net.IPNet
+
+func ParseIPNetSet(specs []string) (IPNetSet, error) {
+	ipnetset := make(IPNetSet)
+	for _, spec := range specs {
+		spec = strings.TrimSpace(spec)
+		_, ipnet, err := net.ParseCIDR(spec)
+		if err != nil {
+			return nil, err
+		}
+		k := ipnet.String() // In case of normalization
+		ipnetset[k] = ipnet
+	}
+	return ipnetset, nil
+}
+
+// StringSlice returns a []string with the String representation of each element in the set.
+// Order is undefined.
+func (s IPNetSet) StringSlice() []string {
+	a := make([]string, 0, len(s))
+	for k := range s {
+		a = append(a, k)
+	}
+	return a
+}
+
+// Equal checks if to IPNetSets are equal (ignoring order)
+func (l IPNetSet) Equal(r IPNetSet) bool {
+	if len(l) != len(r) {
+		return false
+	}
+
+	for k := range l {
+		_, found := r[k]
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+// Len returns the size of the set.
+func (s IPNetSet) Len() int {
+	return len(s)
+}
+
+// GetSourceRangeAnnotations verifies and parses the LBAnnotationAllowSourceRange annotation from a service,
+// extracting the source ranges to allow, and if not present returns a default (allow-all) value.
+func GetSourceRangeAnnotations(annotation ServiceAnnotation) (IPNetSet, error) {
```

Comment:
I still see no value in the `ServiceAnnotation` type - I think we should get rid of it and just call it `map[string]string`